applications/GeoMechanicsApplication/custom_processes/geo_extrapolate_integration_point_values_to_nodes_process.h

```diff
@@ -0,0 +1,131 @@
+    using IndexType    = std::size_t;
+
+    KRATOS_CLASS_POINTER_DEFINITION(GeoExtrapolateIntegrationPointValuesToNodesProcess);
+
+    explicit GeoExtrapolateIntegrationPointValuesToNodesProcess(Model& rModel,
+                                                                Parameters ThisParameters = Parameters(R"({})"));
+    explicit GeoExtrapolateIntegrationPointValuesToNodesProcess(ModelPart& rMainModelPart,
+                                                                Parameters rParameters = Parameters(R"({})"));
+
+    ~GeoExtrapolateIntegrationPointValuesToNodesProcess();
+
+    void                           Execute() override;
+    void                           ExecuteBeforeSolutionLoop() override;
+    void                           ExecuteFinalizeSolutionStep() override;
+    void                           ExecuteFinalize() override;
+    [[nodiscard]] const Parameters GetDefaultParameters() const override;
+    [[nodiscard]] std::string      Info() const override;
+    void                           PrintInfo(std::ostream& rOStream) const override;
+
+private:
+    ModelPart&                                        mrModelPart;
+    std::vector<const Variable<double>*>              mDoubleVariables;
+    std::vector<const Variable<array_1d<double, 3>>*> mArrayVariables;
+    std::vector<const Variable<Vector>*>              mVectorVariables;
+    std::vector<const Variable<Matrix>*>              mMatrixVariables;
+    const Variable<double>&                           mrAverageVariable       = NODAL_AREA;
+    mutable std::map<SizeType, Matrix>                mExtrapolationMatrixMap = {};
+    std::unique_ptr<NodalExtrapolator>                mpExtrapolator;
+    std::map<const Variable<Vector>*, Vector>         mZeroValuesOfVectorVariables;
+    std::map<const Variable<Matrix>*, Matrix>         mZeroValuesOfMatrixVariables;
+
+    void FillVariableLists(const Parameters& rParameters);
+    void InitializeVectorAndMatrixZeros();
+    void InitializeZerosOfVectorVariables(Element&           rFirstElement,
+                                          SizeType           NumberOfIntegrationPoints,
+                                          const ProcessInfo& rProcessInfo);
+    void InitializeZerosOfMatrixVariables(Element&           rFirstElement,
+                                          SizeType           NumberOfIntegrationPoints,
+                                          const ProcessInfo& rProcessInfo);
+    void InitializeVariables();
+    void InitializeAverageVariablesForElements() const;
+
+    template <class T>
+    bool TryAddVariableToList(const std::string& rVariableName, std::vector<const Variable<T>*>& rList) const
+    {
+        const bool variable_is_of_correct_type = KratosComponents<Variable<T>>::Has(rVariableName);
+        if (variable_is_of_correct_type) {
+            auto& thisVariable = KratosComponents<Variable<T>>::Get(rVariableName);
+            rList.push_back(&thisVariable);
```

Comment:
Either rename the local variable to be in line with the KratosStyle Guide:
```suggestion
            auto& this_variable = KratosComponents<Variable<T>>::Get(rVariableName);
            rList.push_back(&this_variable);
```
or eliminate the variable all together:
```suggestion
            rList.push_back(&(KratosComponents<Variable<T>>::Get(rVariableName)));
```

Reply:
Chose the latter option, thanks!